ops/op2/mod.rs

```diff
@@ -281,11 +303,37 @@ fn generate_op2(
       #fast_fn
       #slow_fn
 
-      #[inline(always)]
-      #(#attrs)*
-      #op_fn
+      #op_fn_impl
     }
-  })
+  };
+
+  if *needs_self {
+    let register = format_ident!("register_{name}");
```

Comment:
This shouldn't be necessary -- this code is effectively the same as if it were on the `OpDecl` itself.